src/java_bytecode/java_object_factory.cpp

```diff
@@ -1026,3 +1044,49 @@ void gen_nondet_init(
 
   init_code.append(assignments);
 }
+
+/// Call object_factory with a default (identity) pointer_type_selector
```

Comment:
Does it make sense to actually provide an implementation for the identity selector? (and thus being able to use a reference member instead of a pointer)